grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -1733,6 +1739,42 @@ class InventoryService implements ApplicationContextAware {
         return transactionEntries
     }
 
+    List getTransactionEntriesByInventoryAndProductAndBinLocations(Inventory inventory, Product product, List<Location> binLocations) {
+        def includeDefaultBinLocation = binLocations.contains(null)
+        if (includeDefaultBinLocation) {
+            binLocations = binLocations.findAll {it != null }
+        }
+
+        def criteria = TransactionEntry.createCriteria()
+        def transactionEntries = criteria.list {
+            transaction {
+                eq("inventory", inventory)
+                order("transactionDate", "asc")
+                order("dateCreated", "asc")
+            }
+            inventoryItem { eq("product", product) }
+            or {
+                // Get all transaction entries from transactions with code as INVENTORY or PRODUCT_INVENTORY (with given product)
+                transaction {
+                    transactionType {
+                        inList("transactionCode", [TransactionCode.INVENTORY, TransactionCode.PRODUCT_INVENTORY])
+                    }
+                }
+                // Or get all transactions for that bin location (with given product)
+                or {
+                    if (binLocations) {
+                        inList("binLocation", binLocations)
+                    }
+                    if (includeDefaultBinLocation) {
+                        isNull("binLocation")
+                    }
+                }
```

Comment:
correct me if I'm wrong, but if you anyway do `or` here, to find by `isNull("binLocation")`, was it necessary above (lines 1743-1745) to filter `null` out?
If you hadn't done so, I think you could use here just:
```groovy
if (binLocations) {
  inList("binLocation", binLocations)
}
```
unless `inList` won't match `null`, then I understand everything.

Reply:
@kchelstowski iirc `null` cases won't work with `inList` and `eq` in cases like that, and it requires `isNull`